### PR TITLE
Kfs go samples

### PR DESCRIPTION
**Pull request by bstrzele**

### CODE REVIEW

File: client/go/kserve-api/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+FROM golang:1.18
+ARG http_proxy
```
Comment (jasiu86) [inline client/go/kserve-api/Dockerfile:18]: I'm not convinced if it is required. 
apt-get could working fine without http_proxy


problems with proxy could be machine environment problem.

File: client/go/kserve-api/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+RUN echo "Acquire::http::Proxy \"$http_proxy\";" > /etc/apt/apt.conf.d/proxy.conf
+RUN apt-get update && \
+    apt-get -y install git unzip build-essential autoconf libtool protobuf-compiler libprotobuf-dev
```
Comment (jasiu86) [inline client/go/kserve-api/Dockerfile:22]: please add:
 && rm -rf /var/lib/apt/lists/*

**Review by michalkulakowski — APPROVED** [review]

File: client/go/kserve-api/README.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+```Bash
+./grpc_model_metadata -u localhost:9000
+name:"dummy" versions:"1" platform:"OpenVINO" inputs:{name:"b" datatype:"FP32" shape:1 shape:10} outputs:{name:"a" datatype:"FP32" shape:1 shape:10}
+```
+- Command
+
+```Bash
+./grpc_infer_dummy --help
+Usage of ./grpc_infer_dummy:
+  -m string
```
Comment (ngrozae) [inline client/go/kserve-api/README.md:142]:
```suggestion
  -n string
```

File: client/go/kserve-api/README.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+        Name of model being served.  (default "dummy")
+  -u string
+        Inference Server URL.  (default "localhost:9000")
+  -x string
```
Comment (ngrozae) [inline client/go/kserve-api/README.md:146]:
```suggestion
  -v string
```

File: client/go/kserve-api/README.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+Usage of ./grpc_server_metadata:
+  -u string
+        Inference Server URL.  (default "localhost:9000")
+```
+
+- Usage Example
+
+```Bash
+./grpc_server_metadata -u localhost:9000
+name:"OpenVINO Model Server" version:"2022.3.8fb11b33"
```
Comment (ngrozae) [inline client/go/kserve-api/README.md:91]: output inconsistent with other examples: it's usually:
name:"OpenVINO Model Server"
version:"2022.3.8fb11b33"

File: client/go/kserve-api/README.md [hunk 6]
```diff
@@ -0,0 +1,216 @@
+# KServe API usage samples
+
+OpenVINO Model Server introduced support for [KServe API](https://github.com/kserve/kserve/tree/master/docs/predict-api/v2), including [Triton](https://github.com/triton-inference-server)'s raw format externsion.
```
Comment (dkalinowski) [inline client/go/kserve-api/README.md:3]:
```suggestion
OpenVINO Model Server introduced support for [KServe API](https://github.com/kserve/kserve/tree/master/docs/predict-api/v2), including [Triton](https://github.com/triton-inference-server)'s raw format extension.
```

File: client/go/kserve-api/resnet_labels.txt [hunk 7]
```diff
@@ -0,0 +1,1000 @@
+tench, Tinca tinca
```
Comment (dkalinowski) [inline client/go/kserve-api/resnet_labels.txt:1]: It's not good to add 1k duplicated lines again, same as with java. @michalkulakowski has prepared PR with de-duplication of label files, can you use similar approach?

**Pull request merged**